src/visitable.cpp

```diff
@@ -121,7 +135,7 @@ bool visitable<T>::has_quality( const quality_id &qual, int level, int qty ) con
 template <>
 bool visitable<inventory>::has_quality( const quality_id &qual, int level, int qty ) const
 {
-    long res = 0;
+    int res = 0;
```

Comment:
But they're not consistent, and you previously said that you don't want to try to Fix The World, so it seems safer not to have random extra changes unless you're confident that you're not introducing a new overflow here. (note there is a multiplication below..)